lib/types/SSITypesBuilder.ts

```diff
@@ -11,25 +13,31 @@ import { IVerifiableCredential } from './SSI.types';
 
 export class SSITypesBuilder {
   public static createInternalPresentationDefinitionV1FromModelEntity(p: PdV1): InternalPresentationDefinitionV1 {
+    const presentationDefinitionCopy: PdV1 = JSON.parse(JSON.stringify(p));
+    JsonPathUtils.changePropertyNameRecursively(presentationDefinitionCopy, '_const', 'const');
+    JsonPathUtils.changePropertyNameRecursively(presentationDefinitionCopy, '_enum', 'enum');
     return new InternalPresentationDefinitionV1(
-      p.id,
-      p.input_descriptors,
-      p.format,
-      p.name,
-      p.purpose,
-      p.submission_requirements
+      presentationDefinitionCopy.id,
+      presentationDefinitionCopy.input_descriptors,
+      presentationDefinitionCopy.format,
+      presentationDefinitionCopy.name,
+      presentationDefinitionCopy.purpose,
+      presentationDefinitionCopy.submission_requirements
     );
   }
 
   public static createInternalPresentationDefinitionV2FromModelEntity(p: PdV2): InternalPresentationDefinitionV2 {
+    const presentationDefinitionCopy: PdV2 = JSON.parse(JSON.stringify(p));
```

Comment:
Since we are repeating the same  3 lines in several places I guess it would make sense to create a method out of it. The method needs to return the copy (and also make that clear in the name)

Reply:
done